devicenodegateway creates parent paths when needed

tobsan opened the pull request:
Otherwise mknod will not work if the parent path does not exist.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
devicenodegateway creates parent paths when needed

Otherwise mknod will not work if the parent path does not exist.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

DunderRoffe commented on libsoftwarecontainer/src/gateway/devicenode/devicenodegateway.cpp:
```diff
@@ -65,12 +66,21 @@ ReturnCode DeviceNodeGateway::applySettings()
+        std::string devicePathInContainerOnHost = getContainer()->rootFS() + dev.name;
+        std::string deviceParent = dirname(strdup(devicePathInContainerOnHost.c_str()));
+
+        if (!isDirectory(deviceParent) && isError(createDirectory(deviceParent))) {
```
In order to get better error output, would be nice to catch if `deviceParent` already exists but is not a directory.

DunderRoffe commented on libsoftwarecontainer/src/gateway/devicenode/devicenodegateway.cpp:
```diff
@@ -83,6 +93,8 @@ ReturnCode DeviceNodeGateway::applySettings()
             if (job.wait() != 0) {
```
Would be nice to have a better, or maybe more verbose, way of determine if a job is a failure or not. `job.wait() != 0` is not very self explanatory.

Since all of the code that is run inside a container now use either FunctionJob or CommanJob, I think we should invest in some convenience in these classes. For instance, isSuccess() or isError() would be nice.

Might be out of scope for the pull request though...

tobsan replied:
This is a good idea, but yes, out of scope here I think.

DunderRoffe commented on common/filetoolkitwithundo.h:
```diff
@@ -59,6 +58,18 @@ class FileToolkitWithUndo
+     *
+     * This is useful if one creates a file in some other way and want it deleted later.
+     */
+    void watchFile(const std::string &path);
```
If this method is purely used for deleting files later then I think we can find a better name :)

tobsan replied:
I agree.

Commit:
devicenodegateway: adds further checks on activate

Checks if devices we try to create are already existing, or if
we try to sub-path something that is not a directory

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
